### PR TITLE
MNT: Update feedstock recipe for modern noarch python

### PR DESCRIPTION
**Pull request by matthewfeickert**
* Use `python {{ python_min }}` syntax for the python requirements for noarch python recipes.
   - c.f. https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python
* Use a Jinja2 `set` statement for the `python_min` to allow all the build metadata to be contained in the `recipe/meta.yaml` and override the global `python_min` with `packaging`'s `python_min` of `3.8`.
* Remove `--no-deps` as the build tool (e.g. `conda-build`, `rattler-build`) will specify all required `pip install` options.
   - c.f. https://github.com/conda/grayskull/issues/582
* Use 'pypi.org'.
* Add @matthewfeickert and @ofek as feedstock maintainers.
   - c.f. https://github.com/conda-forge/packaging-feedstock/pull/31#issuecomment-2472019764
* Bump build number.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [N/A] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

**Comment by matthewfeickert** [comment]
The recipe should pass the linter, but the reason for the linter is complaining will be fixed by https://github.com/conda-forge/conda-smithy/pull/2132.

**Comment by matthewfeickert** [comment]
@conda-forge/packaging @conda-forge/core this is ready for review. c.f. https://github.com/conda-forge/packaging-feedstock/pull/31#issuecomment-2472005703

@ofek, please comment or :+1: if you accept being added as a feedstock maintainer.

**Review by ofek — APPROVED** [review]
Looks good to me and I accept.

File: recipe/meta.yaml
Comment (h-vetinari) [inline recipe/meta.yaml]: It would be better to keep these as-is (minus the change to pypi.org, which is necessary). Putting the name behind a variable is useless indirection, and doubly so because it can never be changed.

**Review by h-vetinari — APPROVED** [review]
Thanks for stepping up! @conda-forge/packaging, please let us know if you are OK with some additional support. Otherwise I will merge in ~1 week.

File: recipe/meta.yaml
Comment (matthewfeickert) [reply recipe/meta.yaml]: @h-vetinari I can revise these but I'm following what `grayskull` does at the moment

```console
$ pixi global install grayskull
$ grayskull pypi packaging
$ head -n 11 packaging/meta.yaml 
{% set name = "packaging" %}
{% set version = "24.2" %}

package:
  name: {{ name|lower }}
  version: {{ version }}

source:
  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/packaging-{{ version }}.tar.gz
  sha256: c228a6dc5e932d346bc5739379109d49e8853dd8223571c7c5b55260edc0b97f

```

so maybe we should fix it there first.

Comment (h-vetinari) [reply recipe/meta.yaml]: I know, and I disagree with that too (but not enough that I ever went and tried to get it fixed). 🤷‍♂️

Comment (matthewfeickert) [reply recipe/meta.yaml]: Fixed up in the rebase.

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

**Comment by matthewfeickert** [comment]
Gentle ping to @jakirkham and @nicoddemus.

File: recipe/meta.yaml
Comment (henryiii) [inline recipe/meta.yaml]: Isn't `--no-build-isolation` already specified (via `PIP_NO_BUILD_ISOLATION`) in the environment for all packages?

Comment (matthewfeickert) [reply recipe/meta.yaml]: @henryiii good question, but I have no idea. Where do you think this being set?

Comment (matthewfeickert) [reply recipe/meta.yaml]: Ah, I guess in `conda-build` itself: https://github.com/conda/conda-build/blob/0ad4bb6829b440caae9ae8a4db14ab1ff3788ab7/conda_build/build.py#L3020 ?

I'm now trying to think if there is a downside to doing this / why `grayskull` does this if it isn't needed at all.

Comment (henryiii) [reply recipe/meta.yaml]: PIP_NO_DEPENDENCIES is set too.

I'd guess someone didn't know this was already done.

Comment (matthewfeickert) [reply recipe/meta.yaml]: Oh, seems that you're faster than me and opened https://github.com/conda/grayskull/issues/582 before I could open https://github.com/conda/grayskull/issues/583. :)

**Review by nicoddemus — APPROVED** [review]

File: recipe/meta.yaml
Comment (henryiii) [reply recipe/meta.yaml]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv
```

It sounds like this might be the recommendation going forward.

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

**Comment by matthewfeickert** [comment]
@conda-forge/packaging @conda-forge/core now that we have approval, I've rebased and squashed and rerendered, and this is ready for merge.

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Pull request merged**
